tensorflow_addons/layers/normalizations.py

```diff
@@ -455,36 +452,36 @@ def get_config(self):
         return dict(**base_config, **config)
 
     def _create_input_spec(self, input_shape):
-        ndims = len(tf.TensorShape(input_shape))
+        ndims = len(input_shape)
         for idx, x in enumerate(self.axis):
             if x < 0:
                 self.axis[idx] = ndims + x
 
-        # Validate axes
-        for x in self.axis:
-            if x < 0 or x >= ndims:
-                raise ValueError("Invalid axis: %d" % x)
+        if isinstance(self.axis, list):
+            for x in self.axis:
+                if x < 0 or x >= len(input_shape):
+                    raise ValueError("Invalid axis: %d" % x)
+
+                elif x == 0 or x == -len(input_shape):
+                    raise ValueError(
+                        "You are trying to normalize your batch axis. "
+                        "Use tf.layers.batch_normalization instead."
+                    )
 
-        if len(self.axis) != len(set(self.axis)):
-            raise ValueError("Duplicate axis: %s" % self.axis)
+        else:
+            self.axis = self.channel_idx = -1
```

Comment:
Try not to change the user's params. For 2D tensors, we want the user to explicitly use `self.axis = -1` and `self.channel_idx = -1` (default). If they pass any other value, we should be raising an error instead.